docs/mobile-apps/automated-testing/appium.md

```diff
@@ -1,437 +1,141 @@
-
-<br/>
-
-### 4. Perform Action(s) on the App Elements
-
-Once you've identified the mobile elements you want your test to interact with, the next step is to interact with them. You perform an action on a mobile element by invoking an interaction method on an instance of the WebElement interface.
-
-In this example, we'll test a login button. The WebElement interface declares basic interaction methods including:
-
-* The `sendKeys` method, to enter text.
-* The `clear` method, to clear entered text.
-* The `submit` method, to submit a form.
-
-This example first invokes the `sendKeys` method to enter text in the username and password elements, and then invokes the `submit` method to submit the login form. Enter a username and password:
-
-```java
-emailInput.sendKeys("SauceIsAwesome@email.com");
+curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
+--request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
+--form 'payload=@"/Users/lindsaywalker/Git/Example_Tests/iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"' \
+--form 'name="iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"'
 ```
 
-The `submit` method can be invoked either on any text input element on a form, or on the form element itself. Submit Text Element:
+### Step 4: Set Up Capabilities
+Update your Capabilities to run on Sauce Labs Emulators to have the minimum required Appium Capabilities to run an iOS App test:
 
-```java
-emailInput.submit();
 ```
-
-<br/>
-
-### 5. Anticipate App Response to the Action
-
-When you click a **Submit** button, you know that you have to wait a second or two for your action to reach the server, and for the server to respond, before you do anything else. You want to anticipate the right response (i.e., successful login).
-
-If you're trying to test the response, and what happens afterwards, then you need to build that waiting time into your test. Otherwise, the test might fail because the elements that are expected for the next step haven't loaded into the browser you.
-
-The WebDriver API supports two basic techniques for anticipating browser response by waiting: ___implicit waits___ and ___explicit waits___.  
-
-#### **Implicit Waits Strategy**
-
-_Implicit waits_ set a maximum time that the Appium server will continue trying to find an element. Using implicit waits is not recommended because application response times are not definitely predictable and fixed elapsed times are not applicable to all interactions. Using explicit waits requires more technical sophistication, but is a Sauce Labs best practice.
-
-This example below illustrates how you could use an implicit wait to anticipate web browser response after submitting the login form:
-
-```java
-driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+String appName = "iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa";
```

Comment:
- An Android emulator or real device needs to have an `.apk`-file
- An iOS simulator needs to have an `.app` file which needs to be zipped for the Sauce Labs amulators
- An iOS real device needs to have an `.ipa`-file

The above section is messing the three mentioned points above up into

`Update your Capabilities to run on Sauce Labs Emulators to have the minimum required Appium Capabilities to run an iOS App test`

with a cap like this

`iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa` 

Can we explain this properly in the docs and also provide a correct example?

Reply:
Updated step 3 to only say .ipa, and added a note about diff file types:

### Step 3: Set Up Your App
Use your own app, or grab the latest [Swag Labs App](https://github.com/saucelabs/sample-app-mobile/releases).

On the [Sauce Labs App](https://app.saucelabs.com/live/app-testing), you can upload an .ipa file, or use the API for [Mobile Application Storage](https://docs.saucelabs.com/mobile-apps/app-storage/index.html#uploading-apps-via-rest-api).

The easiest way to do this is to set environment variables for your Sauce username & access key, then use this command:

```
curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
--request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
--form 'payload=@"/Users/lindsaywalker/Git/Example_Tests/iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"' \
--form 'name="iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"'
```
:::note
Different file types are required for testing in different environments:
* An Android emulator or real device needs to have an .apk-file
* An iOS simulator needs to have an .app file which needs to be zipped for the Sauce Labs emulators (.zip file upload)
* An iOS real device needs to have an .ipa-file
:::